gluten-data/src/main/scala/org/apache/spark/sql/execution/utils/GlutenExecUtil.scala

```diff
@@ -182,15 +167,9 @@ object GlutenExecUtil {
         rdd.mapPartitionsWithIndexInternal(
           (_, cbIter) =>
             cbIter.map { cb =>
-              val acb = ArrowColumnarBatches
-                .ensureLoaded(ArrowBufferAllocators.contextInstance(),
-                  cb)
-              (0 until cb.numCols).foreach(
-                acb.column(_)
-                  .asInstanceOf[ArrowWritableColumnVector]
-                  .getValueVector
-                  .setValueCount(acb.numRows))
-              (0, acb)
+              (0, ArrowColumnarBatches.ensureOffloaded(
+                ArrowBufferAllocators.contextInstance(),
```

Comment:
Analyze command `ANALYZE TABLE fact_stats COMPUTE STATISTICS FOR COLUMNS store_id` in test suite `GlutenDynamicPartitionPruningV1SuiteAEOff` will generate `ArrowWritableColumnVector`
```
genShuffleDependency:129, GlutenExecUtil$ (org.apache.spark.sql.execution.utils)
genShuffleDependency:182, GlutenSparkPlanExecApi (io.glutenproject.backendsapi.glutendata)
prepareShuffleDependency:330, ColumnarShuffleExchangeExec$ (org.apache.spark.sql.execution)
columnarShuffleDependency$lzycompute:224, ColumnarShuffleExchangeAdaptor (org.apache.spark.sql.execution)
columnarShuffleDependency:210, ColumnarShuffleExchangeAdaptor (org.apache.spark.sql.execution)
mapOutputStatisticsFuture$lzycompute:200, ColumnarShuffleExchangeAdaptor (org.apache.spark.sql.execution)
mapOutputStatisticsFuture:196, ColumnarShuffleExchangeAdaptor (org.apache.spark.sql.execution)
$anonfun$submitShuffleJob$1:68, ShuffleExchangeLike (org.apache.spark.sql.execution.exchange)
apply:-1, 1817615535 (org.apache.spark.sql.execution.exchange.ShuffleExchangeLike$$Lambda$3410)
$anonfun$executeQuery$1:223, SparkPlan (org.apache.spark.sql.execution)
apply:-1, 796240305 (org.apache.spark.sql.execution.SparkPlan$$Lambda$1926)
withScope:151, RDDOperationScope$ (org.apache.spark.rdd)
executeQuery:220, SparkPlan (org.apache.spark.sql.execution)
submitShuffleJob:68, ShuffleExchangeLike (org.apache.spark.sql.execution.exchange)
submitShuffleJob$:67, ShuffleExchangeLike (org.apache.spark.sql.execution.exchange)
submitShuffleJob:166, ColumnarShuffleExchangeAdaptor (org.apache.spark.sql.execution)
shuffleFuture$lzycompute:170, ShuffleQueryStageExec (org.apache.spark.sql.execution.adaptive)
shuffleFuture:170, ShuffleQueryStageExec (org.apache.spark.sql.execution.adaptive)
doMaterialize:172, ShuffleQueryStageExec (org.apache.spark.sql.execution.adaptive)
materialize:82, QueryStageExec (org.apache.spark.sql.execution.adaptive)
$anonfun$getFinalPhysicalPlan$5:255, AdaptiveSparkPlanExec (org.apache.spark.sql.execution.adaptive)
$anonfun$getFinalPhysicalPlan$5$adapted:253, AdaptiveSparkPlanExec (org.apache.spark.sql.execution.adaptive)
apply:-1, 1686386255 (org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec$$Lambda$3408)
foreach:943, Iterator (scala.collection)
foreach$:943, Iterator (scala.collection)
foreach:1431, AbstractIterator (scala.collection)
foreach:74, IterableLike (scala.collection)
foreach$:73, IterableLike (scala.collection)
foreach:56, AbstractIterable (scala.collection)
$anonfun$getFinalPhysicalPlan$1:253, AdaptiveSparkPlanExec (org.apache.spark.sql.execution.adaptive)
apply:-1, 1729527695 (org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec$$Lambda$3354)
withActive:775, SparkSession (org.apache.spark.sql)
getFinalPhysicalPlan:225, AdaptiveSparkPlanExec (org.apache.spark.sql.execution.adaptive)
withFinalPlanUpdate:367, AdaptiveSparkPlanExec (org.apache.spark.sql.execution.adaptive)
executeTake:344, AdaptiveSparkPlanExec (org.apache.spark.sql.execution.adaptive)
computeColumnStats:267, CommandUtils$ (org.apache.spark.sql.execution.command)
// analyze command may get ArrowWritableVector in GlutenShuffleExecUtil
analyzeColumnInCatalog:118, AnalyzeColumnCommand (org.apache.spark.sql.execution.command)
run:51, AnalyzeColumnCommand (org.apache.spark.sql.execution.command)
sideEffectResult$lzycompute:75, ExecutedCommandExec (org.apache.spark.sql.execution.command)
sideEffectResult:73, ExecutedCommandExec (org.apache.spark.sql.execution.command)
executeCollect:84, ExecutedCommandExec (org.apache.spark.sql.execution.command)
$anonfun$applyOrElse$1:97, QueryExecution$$anonfun$eagerlyExecuteCommands$1 (org.apache.spark.sql.execution)
apply:-1, 1483842690 (org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1$$Lambda$1527)
$anonfun$withNewExecutionId$5:103, SQLExecution$ (org.apache.spark.sql.execution)
apply:-1, 922887623 (org.apache.spark.sql.execution.SQLExecution$$$Lambda$1535)
withSQLConfPropagated:163, SQLExecution$ (org.apache.spark.sql.execution)
$anonfun$withNewExecutionId$1:90, SQLExecution$ (org.apache.spark.sql.execution)
apply:-1, 694580932 (org.apache.spark.sql.execution.SQLExecution$$$Lambda$1528)
withActive:775, SparkSession (org.apache.spark.sql)
withNewExecutionId:64, SQLExecution$ (org.apache.spark.sql.execution)
applyOrElse:97, QueryExecution$$anonfun$eagerlyExecuteCommands$1 (org.apache.spark.sql.execution)
applyOrElse:93, QueryExecution$$anonfun$eagerlyExecuteCommands$1 (org.apache.spark.sql.execution)
$anonfun$transformDownWithPruning$1:481, TreeNode (org.apache.spark.sql.catalyst.trees)
apply:-1, 1325077489 (org.apache.spark.sql.catalyst.trees.TreeNode$$Lambda$1363)
withOrigin:82, CurrentOrigin$ (org.apache.spark.sql.catalyst.trees)
transformDownWithPruning:481, TreeNode (org.apache.spark.sql.catalyst.trees)
org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDownWithPruning:30, LogicalPlan (org.apache.spark.sql.catalyst.plans.logical)
transformDownWithPruning:267, AnalysisHelper (org.apache.spark.sql.catalyst.plans.logical)
transformDownWithPruning$:263, AnalysisHelper (org.apache.spark.sql.catalyst.plans.logical)
transformDownWithPruning:30, LogicalPlan (org.apache.spark.sql.catalyst.plans.logical)
transformDownWithPruning:30, LogicalPlan (org.apache.spark.sql.catalyst.plans.logical)
transformDown:457, TreeNode (org.apache.spark.sql.catalyst.trees)
eagerlyExecuteCommands:93, QueryExecution (org.apache.spark.sql.execution)
commandExecuted$lzycompute:80, QueryExecution (org.apache.spark.sql.execution)
commandExecuted:78, QueryExecution (org.apache.spark.sql.execution)
<init>:219, Dataset (org.apache.spark.sql)
$anonfun$ofRows$2:99, Dataset$ (org.apache.spark.sql)
apply:-1, 287985825 (org.apache.spark.sql.Dataset$$$Lambda$3170)
withActive:775, SparkSession (org.apache.spark.sql)
ofRows:96, Dataset$ (org.apache.spark.sql)
$anonfun$sql$1:618, SparkSession (org.apache.spark.sql)
apply:-1, 616421782 (org.apache.spark.sql.SparkSession$$Lambda$3161)
withActive:775, SparkSession (org.apache.spark.sql)
// 
sql:613, SparkSession (org.apache.spark.sql)
$anonfun$sql$1:231, SQLTestUtilsBase (org.apache.spark.sql.test)
apply:-1, 1838459238 (org.apache.spark.sql.test.SQLTestUtilsBase$$Lambda$3160)
beforeAll:151, DynamicPartitionPruningSuiteBase (org.apache.spark.sql)
org$apache$spark$sql$GlutenSQLTestsTrait$$super$beforeAll:35, GlutenDynamicPartitionPruningSuiteBase (org.apache.spark.sql)
beforeAll:51, GlutenSQLTestsTrait (org.apache.spark.sql)
beforeAll$:49, GlutenSQLTestsTrait (org.apache.spark.sql)
beforeAll:42, GlutenDynamicPartitionPruningSuiteBase (org.apache.spark.sql)
liftedTree1$1:212, BeforeAndAfterAll (org.scalatest)
run:210, BeforeAndAfterAll (org.scalatest)
run$:208, BeforeAndAfterAll (org.scalatest)
run:62, SparkFunSuite (org.apache.spark)
run:45, SuiteRunner (org.scalatest.tools)
$anonfun$doRunRunRunDaDoRunRun$13:1320, Runner$ (org.scalatest.tools)
$anonfun$doRunRunRunDaDoRunRun$13$adapted:1314, Runner$ (org.scalatest.tools)
apply:-1, 1783568981 (org.scalatest.tools.Runner$$$Lambda$168)
foreach:431, List (scala.collection.immutable)
doRunRunRunDaDoRunRun:1314, Runner$ (org.scalatest.tools)
$anonfun$runOptionallyWithPassFailReporter$24:993, Runner$ (org.scalatest.tools)
$anonfun$runOptionallyWithPassFailReporter$24$adapted:971, Runner$ (org.scalatest.tools)
apply:-1, 1848415041 (org.scalatest.tools.Runner$$$Lambda$65)
withClassLoaderAndDispatchReporter:1480, Runner$ (org.scalatest.tools)
runOptionallyWithPassFailReporter:971, Runner$ (org.scalatest.tools)
run:798, Runner$ (org.scalatest.tools)
run:-1, Runner (org.scalatest.tools)
runScalaTest2or3:38, ScalaTestRunner (org.jetbrains.plugins.scala.testingSupport.scalaTest)
main:25, ScalaTestRunner (org.jetbrains.plugins.scala.testingSupport.scalaTest)
```